### PR TITLE
Use DCR Image optimizer for YoutubeAtom

### LINKED ISSUE #8833: Use DCRs image optimizer to optimize images used by the YoutubeAtom
Part of #8541 

After we migrate the YoutubeAtom over to DCR we should try to use the Image Optimizer that DCR provides to create optimal image sizes for the YoutubeAtomOverlay

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

Generate our own srcsets for YoutubeAtom images instead of relying on the ones that Frontend generates.

I've opted to use the same image breakpoints that a regular Image element uses if it was a main media element and has the "inline" role. Theres a couple of future optimizations that could be done I think

- If the YoutubeAtom is part of a Card (shown on a front) we can probably reduce the image size even more.
- If the YoutubeAtom isn't in a main media position we can also probably reduce the image size more.

Closes #8833

## Why?

To improve user experience by using modern compression technology (AVIF) and reducing the amount of data used to render images.

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before_desktop][] | ![after_desktop][] |
| ![before_tablet][] | ![after_tablet][] |
| ![before_mobile][] | ![after_mobile][] |

[after_desktop]: https://github.com/guardian/dotcom-rendering/assets/21217225/5e467596-6815-4007-89ff-1789ede165a1
[after_tablet]: https://github.com/guardian/dotcom-rendering/assets/21217225/5aa8cd00-8ced-4d9e-93a8-4d034d654123
[after_mobile]: https://github.com/guardian/dotcom-rendering/assets/21217225/34779433-d7c0-4ca6-b0ac-a042940c6b05
[before_desktop]: https://github.com/guardian/dotcom-rendering/assets/21217225/3b6041e6-f83d-49d3-9cf9-5579cfb25655
[before_tablet]: https://github.com/guardian/dotcom-rendering/assets/21217225/64b061d1-38d8-4097-bcea-aa9a43da2397
[before_mobile]: https://github.com/guardian/dotcom-rendering/assets/21217225/58b36841-55dd-4917-bf61-5709dab4e705

| Before (Size)      | After (Size)     |
| ----------- | ---------- |
| 69KB | 22KB |
| 28KB | 16KB |

### CODE REVIEW

File: dotcom-rendering/src/components/Card/Card.tsx [hunk 1]
```diff
@@ -459,7 +459,6 @@ export const Card = ({
 										format={format}
 										isMainMedia={true}
 										hideCaption={true}
-										role="inline"
```
Comment (AshCorr) [inline dotcom-rendering/src/components/Card/Card.tsx:462]: We actually never use any role other than "inline" in DCR for the YoutubeAtom. I've deleted a lot of the code around deciding the best image size based on role type in this PR.

**Review by arelra — APPROVED** [review]
✨

File: dotcom-rendering/src/components/Card/Card.tsx (hunk 1 above)
Comment (paperboyo) [reply dotcom-rendering/src/components/Card/Card.tsx:462]: Yeah, super-sadly and absurdly, one can weigh custom embed in Composer, but not atoms. We will get to [this card](https://trello.com/c/yeFnbhkO/205-weighting-of-atoms-in-composer) from 2017 one day, I’m sure…

**Comment by paperboyo** [comment]
🎉  This is brilliant, thanks so much! And [a long time](https://github.com/guardian/frontend/issues/11754#issuecomment-433705157) coming.

To the future optimisations…

> If the YoutubeAtom is part of a Card (shown on a front)

Cards on fronts can be both smaller and much bigger than inlined Main media. Ideally, the image requests should follow. But I’m told images don’t know what size cards they are in (?), so I suppose we would need an array of all possible sizes for browsers to pick from, like we have for normal images.

> If the YoutubeAtom isn't in a main media position we can also probably reduce the image size more.

Main media is same width as Inline.

The only non-inline weightings in articles currently I can think of would be Immersive and Numbered list (which is Showcase by default). Both allowed to carry Media atoms in Composer Main media, but not used by anyone for that purpose (haven’t asked CAPI if odd ones don’t actually exist, though!).

Some loose points (super-sorry if preaching to the choir, but excited about this):

- On non Video pages, the image to actually use for Main media atom should be [a Trail pic from article page](https://github.com/guardian/frontend/pull/20637) and not an atom Poster frame (could be this `overrideImage`, but I’m code-blind)
- The question of should it be Trail pic or Poster on front cards is kinda fluid IIRC (DCR bit less predictable than frontend, but frontend possibly also not 100% correct, even forgetting that I’m not sure anyone knows what correct looks like; safest: imitate frontend for now IMHO). It well may be that one day someone decides that Media atoms should use Poster frame **only** in-body and on Video pages and it should be **article’s** Trail pic at all other times…
- other images do not carry `height` (just `width`). Is it better to have it here?
- hopefully Media atoms have had `master` assets since the very beginning, I hope `getLargestImageSize` will always fetch those
- lastly, just coz I’m code-blind: will this generate a srcset with both lower quality/higher res dpr2 and higher quality/lower res dpr1 assets, like all other imagery?

**Comment by AshCorr** [comment]
> But I’m told images don’t know what size cards they are in

They do to a degree, but I think 

> On non Video pages, the image to actually use for Main media atom should be https://github.com/guardian/frontend/pull/20637 and not an atom Poster frame (could be this overrideImage, but I’m code-blind)

Yea I'm not too sure how this is getting to DCR. I imagine it probably is overrideImage, these 2 props are the way the image url is passed to DCR so it has to be one of them.

> other images do not carry height (just width). Is it better to have it here?

Probably not. Going to avoid touching that in this PR though, I think its complicated enough as is!

> lastly, just coz I’m code-blind: will this generate a srcset with both lower quality/higher res dpr2 and higher quality/lower res dpr1 assets, like all other imagery?

Yes! It should!

> Cards on fronts can be both smaller and much bigger than inlined Main media. 

They do have some info now on how big the image is, but come to think of it the only situation this would really matter is on those cards where the media element is pushed to the left on Mobile, and i think it'd be a bit mad to be using a media atom for those.

**Pull request merged**